src/pyhf/workspace.py

```diff
@@ -295,6 +297,16 @@ def __repr__(self):
         """Representation of the Workspace."""
         return object.__repr__(self)
 
+    @property
+    def hash(self):
+        """
+        A :obj:`dict` mapping the hashing algorithms defined in :attr:`~pyhf.workspace.Workspace.hash_algs` to the computed hash value for the workspace.
+        """
+        return {
+            hash_alg: utils.hash(self, algorithm=hash_alg)
+            for hash_alg in Workspace.hash_algs
+        }
+
```

Comment:
do we need this to be a method of the workspace. I think this might be used that keeping it a standlone method suffices.

```
pyhf.utils.hash(wspace)
```

Reply:
This provides the quick utility to get something that can plug right in the patchset -- but it's a fair point. I could promote this directly to having a CLI option instead, so that it's easier for @danikam's work.